lib/api/filter.rb

```diff
@@ -120,24 +122,16 @@ def single_expression(field, operator, value)
     end
 
     def split_filter_string(filter)
-      operator = nil
-      operators_from_longest_to_shortest = OPERATORS.keys.sort_by(&:size).reverse
-      filter.size.times do |i|
-        operator = operators_from_longest_to_shortest.detect do |o|
-          o == filter[(i..(i + o.size - 1))]
-        end
-        break if operator
-      end
+      filter_match = OPERATOR_REGEXP.match(filter)
+      filter_attr, operator, filter_value = filter_match[1..3] if filter_match
 
-      if operator.blank?
+      if operator.blank? || !OPERATORS.key?(operator)
         raise BadRequestError, "Unknown operator specified in filter #{filter}"
       end
 
-      filter_attr, _op, filter_value = filter.partition(operator)
-      filter_value.strip!
-      filter_attr.strip!
       *associations, attr_name = filter_attr.split(".")
       [MiqExpression::Field.new(model, associations, attr_name), operator, filter_value]
+      # [associations, attr_name, operator, filter_value]
```

Comment:
I don't understand this comment.

Reply:
I dropped.

to make testing easier, I had changed this to the return value.
And I'm still wondering about it.

But to minimize change, probably best to just leave this method as is